### ISSUE TITLE
Error when creating rtdose file, (pinnacle) 

### ISSUE THREAD

**Issue by AviBit** [pymedphys/pymedphys#1706]
Hello, 
who can assist or recommend who should i contact for paid help on this error when exporting Dose file 

pymedphys\_experimental\pinnacle\rtdose.py", line 338, in convert_dose
    value = struct.unpack(">f", data_element)[0]
struct.error: unpack requires a buffer of 4 bytes

**Comment by crcrewso** [comment]
Paid help? I don't think anyone is currently offering paid support for this code. 

Could you give us a few more details:

- Operating system
- Python version
- Pinnacle version
- Do you know anything about the pinnacle dose?
  - Dimensions?
  - Units?
- Any additional information you could add would be a huge help

**Comment by AviBit** [comment]
Sorry im so lost in Python that i need any help i can get. 

-Pinnacle v9.8
-Windows server 
-Python 3.10

i can upload/share all the files for this sample, will anonimize the PHI before if this will help?

**Comment by crcrewso** [comment]
Let's try the simplest thing first. I'm not sure how well tested the Pinnacle tooling was with Python 3.10, and I definitely know that I haven't tested this tooling with any version of Pinnacle since their transition to the Windows codebase. 

Could you try installing python 3.9 and then retrying the code? 
Could you also explain what you're trying to accomplish? I thought the more modern versions of Pinnacle had much better DICOM tooling then the historic versions this tool was built to support.

**Comment by AviBit** [comment]
im trying to convert the old pinnacle files to the dicom version 
i just copy the files from the Sun solaris server to a windows server to run the tool 

on closer look most of the errors i get when the files plan.Trial.binaryXXX are 0-256kb
i only found one case so far where the plan.Trial.binaryXXX files are larger and the error happen 

will look at more failures to see if the pattern is the file size of the plan.Trial.binaryXXX files

**Comment by crcrewso** [comment]
The code definitely wasn't tested against corrupted or empty dose files, sorry we can't help there. 

Good luck

**Comment by AviBit** [comment]
I agree the empty files wont have information 

a diffrent error cause the script to brake and i didn't see it before in the log 

pymedphys\_experimental\pinnacle\rtstruct.py", line 137, in read_points
    roi_contour.ROIDisplayColor = colors[point["Color"]]
KeyError: 'inverse_grey'

any idea about this error?

**Comment by crcrewso** [comment]
Yes. That means the colour isn’t in the index. I need to check my notes on that one.  -- Cody Crewson On July 25, 2022 at 4:23:12 PM, AviBit ***@***.***) wrote: 
I agree the empty files wont have information
a diffrent error cause the script to brake and i didn't see it before in the log
pymedphys_experimental\pinnacle\rtstruct.py", line 137, in read_points
roi_contour.ROIDisplayColor = colors[point["Color"]]
KeyError: 'inverse_grey'
any idea about this error?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/pymedphys/pymedphys/issues/1706#issuecomment-1194707754",
"url": "https://github.com/pymedphys/pymedphys/issues/1706#issuecomment-1194707754",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by SimonBiggs** [comment]
Thanks @crcrewso for taking this one! Appreciated :slightly_smiling_face:
